--- tce-load.orig	2015-05-06 09:23:22.000000000 +0900
+++ tce-load	2015-05-06 09:25:08.000000000 +0900
@@ -85,13 +85,13 @@
 
 copyInstall() {
 	[ -d /mnt/test ] || sudo /bin/mkdir -p /mnt/test
-	sudo /bin/mount $1 /mnt/test -t squashfs -o loop,ro,bs=4096
+	sudo /usr/local/bin/unsquashfs -f -d /mnt/test $1
 	if [ "$?" == 0 ]; then
 		if [ "$(ls -A /mnt/test)" ]; then
 			yes "$FORCE" | sudo /bin/cp -ai /mnt/test/. / 2>/dev/null
 			[ -n "`find /mnt/test/ -type d -name modules`" ] && MODULES=TRUE
 		fi
-		sudo /bin/umount -d /mnt/test
+		sudo rm -rf /mnt/test
 	fi
 }
 
@@ -254,7 +254,7 @@
 		md5sum -cs "$F".md5.txt
 		[ "$?" == 0 ] || fetch_app "$F"
 		[ -f /usr/local/tce.installed/"${F%.tcz}" ] || install "$F"
-	done || exit 1 # subshell cannot exit directly
+	done
 	[ "$DOWNLOAD_ONLY" ] && [ "$ONDEMAND" ] && ondemand "$EXTENSION"
 fi
 
@@ -283,7 +283,7 @@
 				abort_to_saved_dir
 			fi
 		fi
-	done || exit 1 # subshell cannot exit directly
+	done
 fi
 
 cd "$SAVED_DIR"
