--- tce-load.orig	2015-06-01 19:06:25.000000000 +0900
+++ tce-load	2015-06-01 19:05:07.000000000 +0900
@@ -81,13 +81,13 @@
 
 copyInstall() {
 	[ -d /mnt/test ] || sudo /bin/mkdir -p /mnt/test
-	sudo /bin/mount $1 /mnt/test -t squashfs -o loop,ro,bs=4096
+	sudo /usr/local/bin/unsquashfs -f -d /mnt/test $1
 	if [ "$?" == 0 ]; then
 		if [ "$(ls -A /mnt/test)" ]; then
 			yes "$FORCE" | sudo /bin/cp -ai /mnt/test/. / 2>/dev/null
 			[ -n "`find /mnt/test/ -type d -name modules`" ] && MODULES=TRUE
 		fi
-		sudo /bin/umount -d /mnt/test
+		sudo rm -rf /mnt/test
 	fi
 }
 
@@ -266,7 +266,7 @@
 		F="${F##*/}"
 		app_exists "$F" "." || fetch_app "$F"
 		[ -f "$TCEINSTALLED/${F%.*}" ] || install "$F"
-	done || exit 1 # subshell cannot exit directly
+	done || exit 0 # subshell cannot exit directly
 fi
 
 if [ "$TARGETSLOCAL" ]; then
@@ -283,7 +283,7 @@
 				abort_to_saved_dir
 			fi
 		fi
-	done || exit 1 # subshell cannot exit directly
+	done || exit 0 # subshell cannot exit directly
 fi
 
 [ "$BOOTING" ] && exit 0
